src/lib/registration/widgets/package_search.rb

```diff
@@ -141,151 +139,39 @@ def start_search_event?(event)
       # @param text [String] Text to search for
       def search_package(text)
         return unless valid_search_text?(text)
-        @search = ::Registration::PackageSearch.new(text: text)
         # TRANSLATORS: searching for packages
         Yast::Popup.Feedback(_("Searching..."), _("Searching for packages")) do
-          selected_package_ids = selected_packages.map(&:id)
-          @search.packages.each do |pkg|
-            pkg.select! if selected_package_ids.include?(pkg.id)
-          end
+          controller.search(text)
```

Comment:
I expect here call to search as it is not affected by controller and then mark selected packages from controller.